src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,258 @@
+        {
+            Span<byte> rawData = stackalloc byte[6];
+            // Wait for a data to be present
+            if (waitForData)
+            {
+                while (!DataReady)
+                    ;
+            }
+            ReadByteArray(Register.HXL, rawData);
+            // In continous mode, make sure to read the ST2 data to clear up
+            if ((_measurementMode == MeasurementMode.ContinousMeasurement100Hz) ||
+                (_measurementMode == MeasurementMode.ContinousMeasurement8Hz))
+                ReadByte(Register.ST2);
+
+            Vector3 magneto = new Vector3();
+            magneto.X = BinaryPrimitives.ReadInt16LittleEndian(rawData);
+            magneto.Y = BinaryPrimitives.ReadInt16LittleEndian(rawData.Slice(2));
+            magneto.Z = BinaryPrimitives.ReadInt16LittleEndian(rawData.Slice(4));
+            if (OutputBitMode == OutputBitMode.Output16bit)
+            {
+                // From the documentation range is from 32760 which does represent 4912 µT
+                magneto *= 4912.0f / 32760.0f;
+            }
+            else
+            {
+                magneto *= 4912.0f / 8192.0f;
+            }
+            return magneto;
+
+        }
+
+        /// <summary>
+        /// Get the Magetometer data and wait for them to be ready
+        ///         +X
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///    +Z   +Y
+        /// </summary>
+        public Vector3 Magnetometer => ReadMagnetometer(true);
+
+        /// <summary>
+        /// Get or set the device self test mode. 
+        /// If set to true, this creates a magnetic field
+        /// Once you read it, you will have the results of the self test
+        /// 14-bit output(BIT=“0”)  
+        ///          | HX[15:0]        | HY[15:0]        | HZ[15:0] 
+        /// Criteria | -50 =< HX =< 50 | -50 =< HY =< 50 | -800 =< HZ =< -200 
+        /// 16-bit output(BIT=“1”)  
+        ///          | HX[15:0]          | HY[15:0]          | HZ[15:0] 
+        /// Criteria | -200 =< HX =< 200 | -200 =< HY =< 200 | -3200 =< HZ =< -800 
+        /// </summary>
+        public bool SelfTest
+        {
+            get => _selfTest;
+
+            set
+            {
+                byte mode = value ? (byte)0b01000_0000 : (byte)0b0000_0000;
+                WriteRegister(Register.ASTC, mode);
+                _selfTest = value;
+            }
+        }
+
+        /// <summary>
+        /// Select the measurement mode
+        /// </summary>
+        public MeasurementMode MeasurementMode
+        {
+            get => _measurementMode;
+
+            set
+            {
+                byte mode = (byte)((byte)value | ((byte)_outputBitMode << 4));
+                WriteRegister(Register.CNTL, mode);
+                _measurementMode = value;
+                // Need to wait a little bit for the mode to change
```

Comment:
is this delay defined in the spec?

Reply:
Doc says at least 100µs before setting up another mode. So I will adjust to 100µs.